libs/langgraph/tests/test_pregel.py

```diff
@@ -4117,11 +4118,19 @@ def should_continue(data: AgentState) -> str:
         ), "nodes can pass extra data to their cond edges, which isn't saved in state"
         # Logic to decide whether to continue in the loop or exit
         if tool_calls := data["messages"][-1].tool_calls:
-            return [Send("tools", tool_call) for tool_call in tool_calls]
+            return [
+                Send("tools", {"call": tool_call, "my_session": data["session"]})
+                for tool_call in tool_calls
+            ]
         else:
             return END
 
-    def tools_node(tool_call: ToolCall, config: RunnableConfig) -> AgentState:
+    class ToolInput(TypedDict):
+        call: ToolCall
+        my_session: httpx.Client
+
+    def tools_node(input: ToolInput, config: RunnableConfig) -> AgentState:
+        tool_call = input["call"]
```

Comment:
Also assert the `my_session` looks correct?

Reply:
Yep